components/LoadingContainer.js

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo, useState } from 'react';
+import LottieView from 'lottie-react-native';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { View } from '@shoutem/ui';
+import { animations } from '../assets';
+
+function LoadingContainer({
+  animation,
+  children,
+  loading,
+  animationScale,
+  style,
+}) {
+  const [dimensions, setDimensions] = useState({});
+
+  function onLayoutChange(event) {
+    const { width, height } = event.nativeEvent.layout;
+
+    if (width === 0 && height === 0) {
+      return;
+    }
+
+    setDimensions({ width, height });
+  }
+
+  const lottieViewDimensions = useMemo(
+    () => ({
+      width: dimensions.width * animationScale,
+      height: dimensions.height * animationScale,
+    }),
+    [animationScale, dimensions],
+  );
+
+  return (
+    <View style={style.container}>
+      {!loading && <View onLayout={onLayoutChange}>{children}</View>}
+      {loading && (
+        <View style={[dimensions, style.container]}>
```

Comment:
set dimensions as second array member, to avoid faulty style override possibility